src/cloudevent/mocks/helpers.ts

```diff
@@ -28,6 +30,38 @@ export function getBaseCloudEvent<EventType extends CloudEvent<unknown>>(
   } as EventType;
 }
 
+export function makeDataSnapshot(
+  event: database.RawRTDBCloudEvent,
+  instance: string
+): database.DataSnapshot {
+  return new DataSnapshot(event.data, event.ref, undefined, instance);
+}
+
+export function makeChangedDataSnapshot(
+  event: database.RawRTDBCloudEvent,
+  instance: string
+): Change<database.DataSnapshot> {
+  const before = new DataSnapshot(
+    event.data.data,
+    event.ref,
+    undefined,
+    instance
+  );
+  const after = new DataSnapshot(
+    {
+      ...event.data.data,
+      ...event.data.delta,
+    },
```

Comment:
You need to do a recursive merge here. with data of `{"a": {"b": 1, "c": 2}}` and delta of `{"a": {"c": null, "d": 3}}` we should result with the data `{"a": {"b": 1", "d": 3}}`

Reply:
In the SDK we built a helper function called `applyChange` -> https://github.com/firebase/firebase-functions/blob/76e0afd442e141ad0016d01545e5883e27712c3f/src/utils.ts#L26

Reply:
Updated!

Reply:
Maybe we should come up with a pattern for utility functions that we need to export for the sake of the test SDK.

Reply:
True. In this case I reverted the addition because there was already a `makeDataSnapshot` and `makeDataChange` being exposed.